ml-dsa/src/lib.rs

```diff
@@ -899,4 +899,62 @@ mod test {
         sign_verify_round_trip_test::<MlDsa65>();
         sign_verify_round_trip_test::<MlDsa87>();
     }
+
+    fn many_round_trip_test<P>()
+    where
+        P: MlDsaParams,
+    {
+        use rand::Rng;
+
+        const ITERATIONS: usize = 1000;
+
+        let mut rng = rand::thread_rng();
+        let mut seed = B32::default();
+
+        for _i in 0..ITERATIONS {
+            let seed_data: &mut [u8] = seed.as_mut();
+            rng.fill(seed_data);
+
+            let kp = P::key_gen_internal(&seed);
+            let sk = kp.signing_key;
+            let vk = kp.verifying_key;
+
+            let M = b"Hello world";
+            let rnd = Array([0u8; 32]);
+            let sig = sk.sign_internal(&[M], &rnd);
+
+            let sig_enc = sig.encode();
+            let sig_dec = Signature::<P>::decode(&sig_enc).unwrap();
+
+            assert_eq!(sig_dec, sig);
+            assert!(vk.verify_internal(&[M], &sig));
+        }
+    }
```

Comment:
Perhaps consider `proptest` for these sorts of tests.

You can look at https://github.com/RustCrypto/crypto-bigint/tree/master/tests for some examples.

(Or I can try to convert them to proptests as a followup)

Reply:
Note: tried doing this myself and it's a tad tricky due to the use of generics

Reply:
Okay, found an okay enough solution and opened a PR to add proptests: https://github.com/RustCrypto/signatures/pull/896

It also required adding a `Debug` impl for `Signature` as this PR does also.

It was able to find some failures right away, though I did see `many_round_trip_test` find failures with this PR's current state that the proptests did not (but perhaps they just need to be run more).